### ISSUE TITLE
Cannot fetch filter values in dashboard for breakout column from query with aggregation and breakout on implicitly joined field

### ISSUE THREAD

**Issue by kamilmielnik** [metabase/metabase#48613]
https://github.com/user-attachments/assets/8a4caefb-c817-4c1a-980b-95481b7ee6d3

### Repro steps
1. if #48289 is open -> `git checkout 47219-dashboard-drills-tests` 
    - if it's merged -> `git checkout dashboard-filter-columns`
2. New > Question > Orders > Aggregate by Count > Breakout by Created At: Month > Breakout by Product Category
3. Save and add it to a dashboard
4. Add category filter to dashboard and connect it to "Summarize > Category"
5. Save 
6. Click the filter

GET `/api/dashboard/:id/params/:id/values` fails with HTTP 500
```
Error compiling query: Cannot determine the source table or query for Field clause [:field \"PRODUCTS__via__PRODUCT_ID__CATEGORY\" {:base-type :type/Text}]
```

**Comment by kamilmielnik** [comment]
It does not seem that it's field refs bug because it works in query builder filter modal.

**Comment by kamilmielnik** [comment]
Closed by #50120
